nautobot/docs/models/extras/dynamicgroup.md

```diff
@@ -1,7 +1,319 @@
+
+- **Restrict (`AND`)** - The **Restrict** operator performs a set _intersection_ on the queryset, and is equivalent to a Boolean `AND`. Any objects matching the child filter are _restricted_ (aka _intersected_) with the preceding filter. All filter criteria must match between the filters for a member object to be included in the resultant filter.
+- **Include (`OR`)** - The **Include** operator performs a set _union_ on the queryset, and is equivalent to a Boolean `OR`. Any objects matching the child filter are _included_ (aka _unioned_) with the preceding filter. Any filter criteria may match between the filters for member objects to be included in the resultant filter.
+- **Exclude (`NOT`)** - The **Exclude** operator performs a set _difference_ on the queryset, and is equivalent to a Boolean `NOT`. Any objects matching the child filter are _excluded_ (aka _differenced_) from the preceding filter. Any matching objects from the child filter will be negated from the members of the resultant filter.
+
+The following table maps the Nautobot **operator** to the corresponding set operations:
+
+| Operator | Set Operation | Boolean | Description                                                  |
+| -------- | ------------- | ------- | ------------------------------------------------------------ |
+| Restrict | Intersection  | AND     | All criteria from parent/child filters _must_ match          |
+| Include  | Union         | OR      | Any criteria from parent/child filters _may_ match           |
+| Exclude  | Difference    | NOT     | All matches from child filter are _negated_ from parent filter |
+
+Any filters provided by the child groups are used to filter the members from the parent group using one of the three operators: **Restrict (AND)**, **Include (OR)**, or **Exclude (NOT)**. This allows for logical parenthetical grouping of nested groups by the operator you choose for that child group association to the parent.
+
+Using the example group hierarchy from above, let's apply operators and explain how it would work:
+
+```no-highlight
+parent {filter: None}
+- first-child {weight: 10, operator: intersection, filter: site=ams01}
+- second-child {weight: 20, operator: union, fitler: site=ang01}
+- third-child {weight: 30, operator: difference, filter: None}
+  - nested-child {weight: 10, operator: intersection, filter: status=active}
+```
+
+Logically, the filter will be expressed like so using the hierarchy above:
+
+```no-highlight
+((first-child OR second-child) AND (NOT nested-child))
+```
+
+#### How does this work?
+
+First, the filters for direct children for a group will always be included in a parenthetical grouping, separated by the operator. Any _empty_ filters are skipped as they imply a parent group that contains other groups. Therefore the filter of `parent` and the filter of `third-child` are not directly used for generating the filter. The filter from `parent` is passed through from left to right as the "base" filter.
```

Comment:
Maybe instead of talking of skipping empty filters, talk about parent groups each starting from a basis of "all objects"? It's especially confusing as currently worded as it suggests that `parent` is both skipped (since it's an empty filter) and also passed through as a "base" filter, which at first read sounds self-contradictory.

Maybe talk about evaluating from bottom-up instead of left-to-right? I.e. it may be a lot clearer to talk about calculating `third-child` from its children, then applying all three of `first-child`, `second-child`, and the derived `third-child`, versus the current approach of suggesting that `third-child` works fundamentally differently from the other two children.

Reply:
Now I'm confused. Let's pair on this.